convex/lib/withZod.ts

```diff
@@ -16,9 +16,7 @@ import { Id, TableNames } from "../_generated/dataModel";
  * @returns - A Zod object representing a Convex `Id`
  */
 export const zid = <TableName extends TableNames>(tableName: TableName) =>
```

Comment:
I feel like we should just not support `zid` as a custom validator at all because it gives a false sense of security.  E.g. "foo" would validate as a `Id<"users">` because it's a string.

Curious how we feel about pushing folks towards `v.id("users")` instead, which will do the right thing.

Reply:
When you say "push folks toward v.id" do you mean deprecate using Zod? Or is there a way to use `v.id` separately?
The goal of `withZod` is to give users who already use Zod or prefer it, to use it as a drop-in replacement for our `{args, handler}` validation. 
One amazing extension of this would be to translate zod types to convex `args` validation and **also** then do follow-on Zod validation. So Convex could check basic types, and Zod could do any additional things like phone numbers, etc. That would also solve the problem of wanting to use Zod and also wanting the dashboard/ other languages to be able to know the argument shape.
I know @thomasballinger (and others?) talked with the Zod folks about this being possible already.
Then the challenge would just be how to have a "convex ID" string that turns to a string / passthrough validator in Zod and turns into `v.id` internally...

Reply:
If we don't make this work, it's pretty inconvenient to not have something that ends up typed as Id<"tablename>": that will show up as a string param on the client and in the function for every Id...